pkg/controllers/dynakube/deploymentmetadata/deploymentmetadata_test.go

```diff
@@ -42,3 +43,38 @@ func TestFormatKeyValue(t *testing.T) {
 	formattedArgument := formatKeyValue(testKey, testValue)
 	assert.Equal(t, testKey+`=`+testValue, formattedArgument)
 }
+
+func TestGetOneAgentDeploymentType(t *testing.T) {
+	t.Run("deploymentType should be empty", func(t *testing.T) {
+		dynakube := &dynatracev1beta1.DynaKube{}
+		deploymentType := GetOneAgentDeploymentType(*dynakube)
```

Comment:
I think this line is redundant, you can just remove it/
```
assert.Equal(t, "",  GetOneAgentDeploymentType(*dynakube))
```

and in general such tests a good candidates for [table driven tests](https://dave.cheney.net/2019/05/07/prefer-table-driven-tests)